fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2734,7 +2734,7 @@ public void addRebuildRecordCountsJob(List<CompletableFuture<Void>> work) {
         CloseableAsyncIterator<byte[]> cursor = LocalityUtil.getBoundaryKeys(transaction, rangeStart, rangeEnd);
         final boolean hasSplitRecordSuffix = hasSplitRecordSuffix();
         DistinctFilterCursorClosure closure = new DistinctFilterCursorClosure();
-        return new IteratorCursor<>(getExecutor(), cursor)
+        return RecordCursor.fromIterator(getExecutor(), cursor)
                 .flatMapPipelined(
                         result -> new IteratorCursor<>(getExecutor(),
```

Comment:
This should also probably be a `fromIterator` call.

Ideally, I think we'd want to make the constructors for `IteratorCursor` package-private or maybe protected (but for the breaking-ness of the change), I think, to stop errors of this kind.

Reply:
I fixed that other one and checked for more.